### ISSUE TITLE
Open channels

### ISSUE THREAD

**Issue by gareth-ib** [cmgmyr/laravel-messenger#295]
Is there an example of how to use this in public channels rather than user to user?

thanks!

**Comment by gareth-ib** [comment]
but I see no channel concept at the moment in the system, it's just threads on a user

**Comment by Gummibeer** [comment]
What's the difference, for you, between channels and threads?
For me it's just naming and UI - not the DB structure.

**Comment by gareth-ib** [comment]
because the thread would be based on an object that isn't a user, as opposed to threads that are based on a user. and the db structure only allows based on user_ids. I can't just add a trait to a channel class and it'll work cuz the system considers any id to be a user id

**Comment by Gummibeer** [comment]
Ok, we can proceed by one question, one answer or you can provide your whole idea & problem and we can think about it.
Now my solution would be: just make the participants polymorph and you are done. It's also possible with this package cause you can override the model and also change/extend the migration.

**Comment by gareth-ib** [comment]
Basically I picture the system needing a concept of the target subject object, outside of the participants tracking. Or if even participants tracking is needed when threads on that topic are considered public

**Comment by gareth-ib** [comment]
like this comments system tracks to whatever object in question..

https://github.com/actuallymab/laravel-comment/blob/master/database/migrations/0000_00_00_000000_create_comments_table.php#L8

**Comment by Gummibeer** [comment]
Yes, it simply uses a polymorphic relationship - you can add it your own or/and send a PR. 

And if you want to attach threads to another model just add a polymorphic relationship to the thread model.
After this you can do:

```php
$car = Car::where('brand', 'bmw')->first();
$channel = $car->thread; // new polymorphic relationship between thread & another model
$users = $channel->users; // haven't checked the exact method name

dump($users); // contains all "users" assigned via participants - could be a mix of user, car, brand whatever models you allow to be able to join a channel
```

The other way you can do

```php
$channels = Thread::where('is_public', true)->has('subject')->get();
```

This will contain all threads that are public (new column) and have a subject model (new polymorphic relationship).

With this you should be able to achieve whatever you want.

**Comment by gareth-ib** [comment]
okay I forked and PR'd
#297 
it's a start, might have issues? I'm using it right now and continuing it. Hopefully people wanna help with it
